processing/src/main/java/org/apache/druid/segment/column/RowSignature.java

```diff
@@ -392,4 +392,34 @@ public enum Finalization
      */
     UNKNOWN
   }
+
+  /**
+   * Builds a safe {@link RowSignature}.
+   *
+   * The new rowsignature will not contain `null` types - they will be replaced by STRING.
+   */
+  public RowSignature buildSafeSignature(ImmutableList<String> requestedColumnNames)
```

Comment:
Doesn't seem like this really needs to be an `ImmutableList`. A `List` would do.

This method also seems very specific to what `DruidQuery` in the SQL planner needs, and IMO makes more sense as a `static` there.